scripts/backup_data.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+# Backup and restore BirdNET data
+
+my_dir=$HOME/BirdNET-Pi/scripts
+source /etc/birdnet/birdnet.conf
+
+if [ "$EUID" == 0 ]
+  then echo "Please run as a non-root user."
+  exit
+fi
+
+usage() { echo "Usage: $0 -a backup|restore|size -f <backup_file>" 1>&2; exit 1; }
+
+unset -v ACTION
+unset -v ARCHIVE
+unset -v QUIET
+while getopts "a:f:" o; do
+  case "${o}" in
+    a)
+      ACTION=${OPTARG}
+      [ $ACTION == "backup" ] || [ $ACTION == "restore" ] || [ $ACTION == "size" ] || usage
+      ;;
+    f)
+      ARCHIVE=${OPTARG}
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+if [ $ACTION != "size" ]; then
+  [ -z "$ACTION" ] && usage && exit 1
+  [ -z "$ARCHIVE" ] && usage && exit 1
+  [ "$ARCHIVE" == '-' ] && QUIET=1
+fi
+
+PHP_SERVICE=$(systemctl list-unit-files -t service --output json --no-pager php*-fpm.service | jq --raw-output '.[0].unit_file')
+[ -z "$PHP_SERVICE" ] || [ "$PHP_SERVICE" == 'null' ] && echo "Could not determine the php service name, this is most likely a bug." && exit 1
+
+log() {
+  [ -z "$QUIET" ] && echo "$1"
+}
+
+backup_check() {
+  [ "$ARCHIVE" != '-' ] && [ -f "$ARCHIVE" ] && echo "$ARCHIVE already exists" && exit 1
+}
+
+backup() {
+  log "Starting backup"
+  CMD='tar --create -f "$ARCHIVE"'
+  for obj in  "${objects[@]}";do
+    CMD="$CMD -C $(dirname "$obj") $(basename "$obj")"
+  done
+  eval "$CMD"
+  log "Backup done"
+}
+
+estimate_size() {
+  CMD='du -s -c -b '
+  for obj in  "${objects[@]}";do
+    CMD="$CMD $obj"
+  done
+  eval "$CMD | grep total | cut -f 1"
+}
+
+restore_check() {
+  [ ! -f "$ARCHIVE" ] && echo "$ARCHIVE" not found && exit 1
+  arch_list=$(tar --list --exclude="*/*" -f "$ARCHIVE" | sed 's/\///')
+  for obj in  "${objects[@]}";do
+    part2=$(basename "$obj")
+    ! (echo $arch_list | grep -F -q "$part2") && echo corrupted backup file && exit 1
+  done
+}
+
+restore() {
+  log "Starting restore"
+  for obj in  "${objects[@]}";do
+    tar --extract - p -f "$ARCHIVE" -C "$(dirname "$obj")" "$(basename "$obj")"
```

Comment:
There's a space between - and p that shouldn't be there...

Reply:
doh!